presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -175,9 +175,11 @@ private static TypeSignature parseRowTypeSignature(String signature, Set<String>
             else if (c == ' ') {
                 if (bracketCount == 1 && inFieldName) {
                     checkArgument(parameterStart >= 0, "Bad type signature: '%s'", signature);
-                    fieldNames.add(signature.substring(parameterStart, i));
+                    if (parameterStart < i) {
```

Comment:
I would just write it without the spaces. Because this is part of the public API of Presto, I'm reluctant to change it as it could break clients